packages/cli/src/commands/generate/__tests__/helpers.test.js

```diff
@@ -546,13 +546,21 @@ describe('mapPrismaScalarToPagePropTsType', () => {
 })
 
 describe('validateName', () => {
-  it('throws an error if name starts with a slash', () => {
+  it('throws an error if name starts with an invalid character', () => {
     expect(() => helpers.validateName('/')).toThrow()
     expect(() => helpers.validateName('/foo')).toThrow()
+    expect(() => helpers.validateName('.foo')).toThrow()
+    expect(() => helpers.validateName(',foo')).toThrow()
+    expect(() => helpers.validateName('-foo')).toThrow()
   })
   it('does nothing if name is valid', () => {
     expect(() => helpers.validateName('foo')).not.toThrow()
     expect(() => helpers.validateName('foo/')).not.toThrow()
+    expect(() => helpers.validateName('_foo')).not.toThrow()
+    expect(() => helpers.validateName('1foo')).not.toThrow()
     expect(() => helpers.validateName('foo/bar')).not.toThrow()
```

Comment:
@cannikin Thinking about [the problem with creating a page in a sub directory](https://github.com/redwoodjs/redwood/issues/7324#issuecomment-1382158434). I was checking to update the regex pattern in this PR, but this line tests for a `directory/page` and expects no error. 🤷‍♂️ It seems like this would pass the subdirectory creation pattern.

Reply:
So there's no actual error during the generation, but the page itself, when React tries to render it, throws an error because the route function call is invalid syntax `routes.admin / dashboard()`.